lib/private/Settings/DeclarativeManager.php

```diff
@@ -290,18 +292,54 @@ private function getForm(string $app, string $formId): ?IDeclarativeSettingsForm
 	private function getInternalValue(IUser $user, string $app, string $formId, string $fieldId): mixed {
 		$sectionType = $this->getSectionType($app, $fieldId);
 		$defaultValue = $this->getDefaultValue($app, $formId, $fieldId);
+
+		$form = $this->getForm($app, $fieldId);
+		if ($form === null) {
+			$field = $this->getSchemaField($app, $formId, $fieldId);
+		} else {
+			$field = $this->getSchemaField($app, $formId, $fieldId, $form);
+		}
+		$isSensitive = $field !== null && isset($field['sensitive']) && $field['sensitive'];
+
 		switch ($sectionType) {
 			case DeclarativeSettingsTypes::SECTION_TYPE_ADMIN:
-				return $this->config->getAppValue($app, $fieldId, $defaultValue);
+				$value = $this->config->getAppValue($app, $fieldId, $defaultValue);
+				break;
 			case DeclarativeSettingsTypes::SECTION_TYPE_PERSONAL:
-				return $this->config->getUserValue($user->getUID(), $app, $fieldId, $defaultValue);
+				$value = $this->config->getUserValue($user->getUID(), $app, $fieldId, $defaultValue);
+				break;
 			default:
 				throw new Exception('Unknown section type "' . $sectionType . '"');
 		}
+		if ($isSensitive && !empty($value)) {
```

Comment:
```suggestion
		if ($isSensitive && $value !== '') {
```
At this point the value is only a string, right? Then it's much better to check exactly what you want to avoid surprises.